### PR TITLE
planner: don't push down some conditions to projections

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/35623
close https://github.com/pingcap/tidb/issues/37971

Problem Summary:
Some conditions shouldn't be pushed down to projection.

### What is changed and how it works?
Check whether the conditions can be pushed down to projection.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: expression/util.go [hunk 1]
```diff
@@ -382,6 +382,17 @@ func SetExprColumnInOperand(expr Expression) Expression {
 	return expr
 }
 
+// ColumnSubstitute4PPD2P substitutes the columns in filter to expressions in select fields.
+// Only used for predicate push down to projection. Some columns can not be substituted for some reasons.
+// So we should return the bool value to indicate some expressions can not be pushed down.
+// e.g. CREATE TABLE t3(c0 INT, primary key(c0));
+// SELECT v2.c0 FROM (select 1 as c0 from t3) v2 WHERE (v2.c0)like(True);
+// The cond `(v2.c0)like(True)` can not be substituted when the new collation enable. So we shouldn't push the cond down to the projection.
+func ColumnSubstitute4PPD2P(expr Expression, schema *Schema, newExprs []Expression) (bool, Expression) {
```
Comment (AilinKid) [inline expression/util.go:391]: the func name is complicated >.<

**Review by xuyifangreeneyes — APPROVED** [review]

**Comment by Reminiscent** [comment]
THE pr https://github.com/pingcap/tidb/pull/38802 handle the same problem but in a more reasonable way. So we don't need this PR because that PR has changed the code of this PR.

**Pull request merged**